capstone/src/test/java/com/google/sps/servlets/AnnouncementTest.java

```diff
@@ -0,0 +1,140 @@
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalURLFetchServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+/** */
+@RunWith(JUnit4.class)
+public final class AnnouncementTest {
+
+  private static final String ANNOUNCEMENT_PROP = "Announcement";
+  private static final String AUTHOR_PROP = "author";
+  private static final String TIME_PROP = "time";
+  private static final String CONTENT_PROP = "content";
+  private static final String CLUB_PROP = "club";
+
+  private static final String AUTHOR_1 = "kshao@google.com";
+  private static final String AUTHOR_2 = "kakm@google.com";
+  private static final String AUTHOR_3 = "meganshi@google.com";
+
+  private static final String CLUB_1 = "Club 1";
+  private static final String CLUB_2 = "Club 2";
+  private static final String CLUB_3 = "Club 3";
+  private static final String CONTENT_1 = "Cool test comment.";
+
+  private static final long TIME_10 = 10L;
+
+  private AnnouncementsServlet servlet;
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+  private DatastoreService datastore;
+
+  private Entity announcement1;
+  private Entity announcement2;
+
+  private LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+          new LocalDatastoreServiceTestConfig(),
+          new LocalUserServiceTestConfig(),
+          new LocalURLFetchServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    this.servlet = new AnnouncementsServlet();
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    this.datastore = DatastoreServiceFactory.getDatastoreService();
+
+    announcement1 = new Entity(ANNOUNCEMENT_PROP);
+    announcement1.setProperty(AUTHOR_PROP, AUTHOR_1);
+    announcement1.setProperty(CLUB_PROP, CLUB_1);
+    announcement1.setProperty(CONTENT_PROP, CONTENT_1);
+    announcement1.setProperty(TIME_PROP, TIME_10);
+
+    this.datastore.put(announcement1);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void club2HasNoAnnouncements() throws IOException {
```

Comment:
nit: test methods should be testing general behavior, so the name should just be "clubHasNotAnnouncements" instead of specifying the specific club you're using as the example in the test. Same for below